allennlp/predictors/text_classifier.py

```diff
@@ -22,4 +25,15 @@ def _json_to_instance(self, json_dict: JsonDict) -> Instance:
         Runs the underlying model, and adds the ``"label"`` to the output.
         """
         sentence = json_dict["sentence"]
+        if isinstance(self._dataset_reader, StanfordSentimentTreeBankDatasetReader):
```

Comment:
Why do you need this?  Having a check for a particular subclass is _really_ bad practice for this kind of code.

Reply:
the SST dataset expects pre-tokenized text, so if you don't tokenize before, then it will split words up into their individual tokens

Reply:
Maybe check instead for `if not hasattr(self._dataset_reader, 'tokenizer') and not hasattr(self._dataset_reader, '_tokenizer')`.  That way we assume that if you have no tokenizer in your dataset reader, you take pre-tokenized input, and it's not specific to a subclass.

Reply:
This still isn't great, and really is an issue with the dataset readers.

Reply:
ill do that